content/ui/popup.js

```diff
@@ -0,0 +1,197 @@
+
+  chrome.runtime.sendMessage(
+    { action: "logout", password: pass2 },
+    (response) => {
+      if (response && response.success === true) {
+        statusDiv.textContent = "Logged out successfully!";
+        updatePopupContent();
+      } else {
+        statusDiv.textContent = response.error;
+        passwordInput.value = "";
+      }
+    }
+  );
+};
+
+// Function to update popup content based on login status
+const updatePopupContent = async () => {
+  const signUpContainer = document.getElementById("signUpContainer");
+  const kidsContent = document.getElementById("kidsContent");
+  const signInContainer = document.getElementById("signInContainer");
+  await chrome.storage.local.get(["loggedIn"], (data) => {
+
+    updateTimeLeftUI()
+
+    if (data.loggedIn) {
+      // User is logged in
+      signUpContainer.style.display = "none";
+      signInContainer.style.display = "none";
+      kidsContent.style.display = "block";
+    } else if (data.loggedIn === false) {
+      // User is not logged in
+      signUpContainer.style.display = "none";
+      signInContainer.style.display = "block";
+      kidsContent.style.display = "none";
+    } else {
+      signUpContainer.style.display = "block";
+      signInContainer.style.display = "none";
+      kidsContent.style.display = "none";
+    }
+  });
+};
+
+const setSessionTimer = () => {
+  // Get the select element
+  const sessionTimeSelect = document.getElementById("sessionTime");
+
+  // Array of numbers from 1 to 24
+  const hours = Array.from({ length: 24 }, (_, index) => index + 1);
+
+  // Iterate over the array to create options
+  hours.forEach(hour => {
+    const option = document.createElement("option");
+    option.value = hour;
+    option.text = hour + " Hr" + (hour !== 1 ? "s" : ""); // Pluralize "Hour" if hour is not 1
+    sessionTimeSelect.appendChild(option);
+  });
+}
+
+const resizeDropdown = () => {
+  const selectElement = document.getElementById("sessionTime");
+
+  selectElement.addEventListener('mousedown', () => {
+    if (selectElement.options.length > 6) {
+      selectElement.size = 6;
+    }
+  });
+
+  selectElement.addEventListener('change', () => {
+    selectElement.size = 0;
+  });
+
+  selectElement.addEventListener('blur', () => {
+    selectElement.size = 0;
+  });
+};
+
+// Function to fetch timeLeft from local storage and update UI
+const updateTimeLeftUI = () => {
+  let x = setInterval(() => {
+    chrome.storage.local.get("timeLeft", (data) => {
+      let distance = data.timeLeft;
+      if (distance !== undefined) {
+        let hours = Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        let minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
+        document.getElementById("timerDisplay").textContent = "Time Left : " + hours + "h "
+          + minutes + "m ";
+        if (distance <= 0) {
+          clearInterval(x);
+          document.getElementById("timerDisplay").textContent = "expired";
```

Comment:
store the reference to element in a variable. this function take traverses the whole dom to find the element and is hence slow